pkg/template/master_template.go

```diff
@@ -204,12 +208,12 @@ systemd:
           --peer-key-file /etc/etcd/server-key.pem \
           --peer-client-cert-auth=true \
           --advertise-client-urls=https://{{ .Cluster.Etcd.Domain }}:{{ .Etcd.ClientPort }} \
-          --initial-advertise-peer-urls=https://{{ .Etcd.NodeName }}.{{ .BaseDomain }}:2380 \
+          --initial-advertise-peer-urls=http://${DEFAULT_IPV4}:2380 \
```

Comment:
Is there a reason not to listen on the LAN address here and use the hostname instead?

Reply:
Yes this is a problem for AWS, as the DEFAULT IPV4 is not the same ip that is etcd using for connecting

Reply:
if you need this for azure than put it into the ENV as well and in azure you can override it

Reply:
what if we use 0.0.0.0?

Reply:
that is not a "listen address" that is IP that is sent as Peer URL to other etcds

you need to be able to connect to that

Reply:
and definitely not http but https